lib/utils.js

```diff
@@ -50,7 +50,7 @@ exports.validateArg = function validateArg (args, config) {
         break;
 
       case Function:
-        if (toString.call(value) !== '[object Function]') {
+        if (typeof obj !== "undefined" && toString.call(value) !== '[object Function]') {
```

Comment:
I guess my point is that if it's optional, you may want to check for optionality before calling this function?